logging/writer.go

```diff
@@ -1,29 +1,70 @@
 package logging
 
-import "io"
+import (
+	"bytes"
+	"io"
+	"strings"
+	"sync"
+)
 
 // writer is an implementation of io.Writer that writes to a logger.
 type writer struct {
 	l Logger
 	f func(Logger, string)
+
+	mu  sync.Mutex
```

Comment:
Very pedantic consistency suggestion!

```suggestion
	m  sync.Mutex
```